src/basic/log.c

```diff
@@ -1341,6 +1344,14 @@ void log_received_signal(int level, const struct signalfd_siginfo *si) {
                          signal_to_string(si->ssi_signo));
 }
 
+void set_log_syntax_callback(log_syntax_callback_t cb, void *userdata) {
+        assert(cb || log_syntax_callback);
```

Comment:
i figure you mean:

```c
assert(!cb || log_syntax_callback);
```

i.e. when a callback is currently set the only OK parameter is NULL

Reply:
So - 
assert(!cb || log_syntax_callback)
assert(!userdata || log_syntax_callback_userdata)

Reply:
given there's automatic scoped cleanup, so it's possible that clear_log_syntax_callback is called before it is set, I'd assert like: `assert(!log_syntax_callback || !cb)` ie, if it's not currently configured, then it's ok whatever gets passed. But if it is currently configured, then only passing NULL is allowed.